core/interfaces/i_ast_location.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview AST Node and navigation interfaces.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IASTNode');
+goog.provide('Blockly.IASTNodeWithBlock');
+goog.provide('Blockly.IBlocklyActionable');
+
+/**
+ * An AST node interface.
+ * @interface
+ */
+Blockly.IASTNode = function() {};
+
+
+/**
+ * An AST node location that has an associated block.
+ * @interface
+ * @extends {Blockly.IASTNode}
+ */
+Blockly.IASTNodeWithBlock = function() {};
+
+/**
+ * Get the source block associated with this node.
+ * @return {Blockly.Block} The source block.
+ */
+Blockly.IASTNodeWithBlock.prototype.getSourceBlock;
+
+
+/**
+ * An interface for an object that handles Blockly actions.
+ * @interface
+ */
+Blockly.IBlocklyActionable = function() {};
```

Comment:
Hmm, curious why you feel that way about the interface but not the method: `onBlocklyAction`.

Reply:
Yeah, I'm not sure I don't feel that way about `onBlocklyAction`. I think the hope was that Blockly Actions would get used more widely in the future, but it makes the name currently a bit misleading. 

Anyway, fine with the name since it matches `onBlocklyAction` but would at least like a comment explaining it is currently used for keyboard navigation.

Reply:
Expanded on the interface description.